src/notebook/plugin.ts

```
@@ -40,20 +51,66 @@ export
 function resolve(container: Container): void {
   container.resolve({
     requires: [IServicesProvider, IFileOpener],
-    create: (services, opener) => {
-      let handler = new NotebookFileHandler(services.contentsManager);
+    create: (services: IServicesProvider, opener: IFileOpener) => {
+      let handler = new NotebookFileHandler(services.contentsManager, services.notebookSessionManager);
       opener.register(handler);
     }
   });
 }
 
+export
+class SessionStoreMapping {
+  constructor(services: IServicesProvider) {
+    this.services = services;
+  }
+  public services: IServicesProvider;
+}
+
+function messageToModel(msg: IKernelMessage) {  
+  let m: Output = msg.content;
+  let type = msg.header.msg_type;
+  if (type === 'execute_result') {
+    m.output_type = 'display_data';
+  } else {
+    m.output_type = type;
+  }
+  return buildOutputModel(m);
+}
+
+function executeSelectedCell(model: NotebookModel, session: INotebookSession)  {
+  let cell = model.cells.get(model.selectedCellIndex);
+  if (isCodeCell(cell)) {
+    let exRequest = {
+      code: cell.input.textEditor.text
+    };
+    let output = cell.output;
+    console.log(`executing`, exRequest)
+    let ex = session.kernel.execute(exRequest);
+    output.clear(false);
+    ex.onIOPub = (msg => {
+      // TODO: not getting an execute_result message
```

The docs for kernel.js contradict themselves.  https://github.com/jupyter/notebook/blame/master/notebook/static/services/kernels/kernel.js#L700 says silent should default to false, but just below it, https://github.com/jupyter/notebook/blame/master/notebook/static/services/kernels/kernel.js#L710 says silent defaults to true.

Looks like this commit changed it: https://github.com/jupyter/notebook/commit/f306423c7ea3bc60735a8bfb4f5f2cc60dc36b97